client/src/components/Header/Menu.js

```diff
@@ -0,0 +1,230 @@
+          sx: {
+            overflow: 'visible',
+            filter: 'drop-shadow(0px 2px 8px rgba(0,0,0,0.32))',
+            mt: 1.5,
+            '& .MuiAvatar-root': {
+              width: 32,
+              height: 32,
+              ml: -0.5,
+              mr: 1,
+            },
+            '&:before': {
+              content: '""',
+              display: 'block',
+              position: 'absolute',
+              top: 0,
+              right: 14,
+              width: 10,
+              height: 10,
+              bgcolor: 'background.paper',
+              transform: 'translateY(-50%) rotate(45deg)',
+              zIndex: 0,
+            },
+          },
+        }}
+      >
+        {isAuthenticated && (
+          <div className="header__navigation__authcontainer__menu">
+            <div className="header__navigation__authcontainer__menu__profile">
+              <Link
+                className="header__navigation__authcontainer__menu__menuitem__links"
+                to="/userprofile"
+              >
+                <MenuItem
+                  onClick={handleClose}
+                  sx={{
+                    padding: '8px',
+                    width: '180px',
+                    borderBottom: '#00000040 1px solid',
+                  }}
+                >
+                  <AccountCircleOutlined sx={{ fontSize: '2rem' }} />
+                  <span className="header__navigation__authcontainer__menu__menuitem__links__span">
+                    Profile
+                  </span>
+                </MenuItem>
+              </Link>
+            </div>
+          </div>
+        )}
+
+        <div className="header__navigation__authcontainer__menu__menuitem">
+          <Link
+            className="header__navigation__authcontainer__menu__menuitem__links"
+            to="/"
+          >
+            <MenuItem
+              sx={{
+                padding: '8px',
+                width: '180px',
+              }}
+              onClick={handleClose}
+            >
+              <PublicOutlined sx={{ fontSize: '2rem' }} />
+              <span className="header__navigation__authcontainer__menu__menuitem__links__span">
+                Map
+              </span>
+            </MenuItem>
+          </Link>
+
+          <Link
+            className="header__navigation__authcontainer__menu__menuitem__links"
+            to="/about"
+          >
+            <MenuItem
+              sx={{
+                padding: '8px',
+                width: '180px',
+              }}
+              onClick={handleClose}
+            >
+              <InfoOutlined sx={{ fontSize: '2rem' }} />
+              <span className="header__navigation__authcontainer__menu__menuitem__links__span">
+                About
+              </span>
+            </MenuItem>
+          </Link>
+
+          <Link
+            to="/contact"
+            className="header__navigation__authcontainer__menu__menuitem__links"
+          >
+            <MenuItem
```

Comment:
This seems like something that can be reused. Maybe have a react component like
```
const YourNewComponentName = ({ children, title }) => {
return
 <MenuItem
              sx={{
                padding: '8px',
                width: '180px',
              }}
              onClick={handleClose}
            >
              {children}
              <span className="header__navigation__authcontainer__menu__menuitem__links__span">
                {title}
              </span>
            </MenuItem>
 }
```

Reply:
good idea actually, I had trouble with the Menu not anchoring correctly to the button, but the menu items shouldnt be an issue will look into refractoring that